### PR TITLE
Add Findby command

### PR DESCRIPTION
**Pull request by MuhdNurKamal**
Issue #3 
Findby finds contact according to specified field

### CODE REVIEW

**Comment by MuhdNurKamal** [comment]
#3

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -284,6 +304,8 @@ e.g. `edit 2 n/James Lee e/jameslee@example.com`
 e.g. `removegroup g/publicity`
 * *Find* : `find KEYWORD [MORE_KEYWORDS]` +
 e.g. `find James Jacob`
+* *Find* : `findby FIELD KEYWORD [MORE_KEYWORDS]` +
```
Comment (amrut-prabhu) [inline docs/UserGuide.adoc:307]: Change the title from `Find`

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -41,6 +41,29 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         return false;
     }
 
+    /**
+     * Returns true if the {@code sentence} contains the {@code word}.
+     *   Ignores case and finds partial match.
+     *   <br>examples:<pre>
```
Comment (amrut-prabhu) [inline src/main/java/seedu/address/commons/util/StringUtil.java]: remove the tags (<br>, <pre>)?

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 3]
```diff
@@ -41,6 +41,29 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+     *       containsWordIgnoreCase("ABc def", "abc") == true
+     *       containsWordIgnoreCase("ABc def", "DEF") == true
+     *       containsWordIgnoreCase("ABc def", "AB") == true (partial match)
+     *       containsWordIgnoreCase("ABc def", "ABCD") == false
+     *       </pre>
+     * @param sentence cannot be null
+     * @param word cannot be null, cannot be empty, must be a single word
+     */
+    public static boolean partiallyContainsWordIgnoreCase(String sentence, String word) {
```
Comment (amrut-prabhu) [inline src/main/java/seedu/address/commons/util/StringUtil.java:56]: Change `word` to `toFind` or `toSearch`
Change `sentence` to more appropriate name (maybe, `fieldValue`)

File: src/main/java/seedu/address/logic/parser/FindByCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import seedu.address.logic.commands.FindByCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.FieldContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindByCommand object
+ */
+public class FindByCommandParser implements Parser<FindByCommand> {
+
+    private static final String[] fieldTypes = {"name", "email", "phone", "matric", "tag", "group"};
```
Comment (amrut-prabhu) [inline src/main/java/seedu/address/logic/parser/FindByCommandParser.java:16]: Should we change this to e/, p/, t/ etc. for consistency?

File: src/test/java/seedu/address/logic/commands/FindByCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    private FindByCommand prepareCommand(String userInput, String fieldType) {
+        FindByCommand command =
+                new FindByCommand(new FieldContainsKeywordsPredicate(Arrays.asList(userInput.split("\\s+")),
+                        fieldType));
+        command.setData(model, new CommandHistory(), new UndoRedoStack());
+        return command;
+    }
+
+    /**
+     * Asserts that {@code command} is successfully executed, and<br>
```
Comment (amrut-prabhu) [inline src/test/java/seedu/address/logic/commands/FindByCommandTest.java]: remove tags

**Review by amrut-prabhu — COMMENTED** [review]
Suggestions for minor changes to be made in a subsequent PR (also, see above comments).

File: docs/UserGuide.adoc (hunk 1 above)
Comment (amrut-prabhu) [reply docs/UserGuide.adoc:307]: As discussed in class, let's just merge the two commands

File: src/main/java/seedu/address/logic/parser/FindByCommandParser.java (hunk 4 above)
Comment (amrut-prabhu) [reply src/main/java/seedu/address/logic/parser/FindByCommandParser.java:16]: As discussed in class, let's not require fieldTypes.
But, maybe we can make it optional. For example, if you only want to search for "ab" in emails and not names of persons.

**Pull request merged**